### PR TITLE
chore(triple): add doc comments and boundary tests for stream APIs

### PR DESCRIPTION
**Pull request by lizining1231**
### Description
Fixes #3644 (task 9)
This PR adds doc comments for the Triple stream exported APIs, clarifying their behavior, error return conditions, stop semantics, and metadata handling. It also adds 20 boundary tests covering stream close, repeated close, error propagation and empty metadata.

### Changes

- **client_stream.go**
  - `CloseAndReceive`: doc comment enhanced — closes the receive side before returning an error, whether closing the send side or receiving the response fails.
  - `ServerStreamForClient.Receive`: documents the stop semantics — once Receive has returned false, subsequent calls return false without advancing the stream.
  - `ServerStreamForClient.Close`: returns the construction error when the stream could not be constructed.
  - `ServerStreamForClient.Msg`: aligned with the handler side — returns nil if Receive has not been called, and the message may be left unmarshaled after a failed Receive.
  - `BidiStreamForClient.Receive`: returns the underlying receive error unchanged.
- **handler_stream.go**
  - `ClientStream.Receive`: documents the same stop semantics as the client side.
  - `ClientStream.Msg`: documents the failure semantics of the returned message.
  - `ServerStream.Send` / `BidiStream.Send`: fixed the misleading comment — a nil msg does not send an empty message, it commits the response headers without writing a message frame.
  - `BidiStream.ExportableHeader`: clarified that the underlying protocol decides which headers are exported; under the gRPC protocol reserved headers are filtered out (except the whitelisted ones) and keys are lowercased.
- **handler_stream_compat.go**
  - `NewCompatStreamHandler`: new doc comment. Describes the `grpc.ServerStream` adaptation, the metadata and `TripleGoInterfaceName` values injected into the context, the interceptor wrapping order (the interceptor wraps the whole implementation, so it runs before the metadata injection), and the fact that `SetHeader` / `SendHeader` / `SetTrailer` on the adapted stream are no-ops.

**client_stream_test.go (11 new tests):**

| Test | What it verifies |
| --- | --- |
| `TestServerStreamForClientClose` | `Close` delegates to `CloseResponse` and returns its result |
| `TestServerStreamForClientCloseRepeated` | Calling `Close` more than once keeps delegating |
| `TestServerStreamForClientErrPropagation` | `Err` returns the first non-EOF error and nil for a normal end of stream (EOF) |
| `TestServerStreamForClientReceiveStopsAfterError` | Once an error is recorded, subsequent `Receive` calls return false without advancing the connection |
| `TestServerStreamForClientEmptyMetadata` | `ResponseHeader` / `ResponseTrailer` return empty headers |
| `TestBidiStreamForClientReceiveErrorPropagation` | `Receive` returns the underlying connection error unchanged |
| `TestBidiStreamForClientRequestHeaderEmpty` | `RequestHeader` returns an empty header |
| `TestBidiStreamForClientCloseRequestRepeated` | Calling `CloseRequest` more than once keeps delegating |
| `TestBidiStreamForClientCloseResponseRepeated` | Calling `CloseResponse` more than once keeps delegating |
| `TestClientStreamForClientCloseAndReceiveClosesResponseOnCloseRequestError` | `CloseAndReceive` closes the receive side before returning the error when closing the send side fails |
| `TestClientStreamForClientCloseAndReceiveClosesResponseOnReceiveError` | `CloseAndReceive` closes the receive side before returning the error when receiving the response fails |

**handler_stream_test.go (9 tests):**

| Test | What it verifies |
| --- | --- |
| `TestClientStreamReceiveStopsAfterError` | Once an error is recorded, subsequent `Receive` calls return false without advancing the connection |
| `TestClientStreamErrPropagation` | `Err` returns the first non-EOF error and nil for a normal end of stream (EOF) |
| `TestClientStreamMsgNil` | `Msg` returns nil when `Receive` has not been called |
| `TestClientStreamRequestHeaderEmpty` | `RequestHeader` returns an empty header |
| `TestServerStreamSendNil` | `Send` delegates a nil msg to the underlying connection unchanged |
| `TestServerStreamResponseHeaderEmpty` | `ResponseHeader` / `ResponseTrailer` return empty headers |
| `TestBidiStreamReceiveErrorPropagation` | `Receive` returns the underlying connection error unchanged |
| `TestBidiStreamSendNil` | `Send` delegates a nil msg to the underlying connection unchanged |
| `TestBidiStreamExportableHeader` | `ExportableHeader` forwards the underlying connection's metadata unchanged |

### Checklist
- [x] I confirm the target branch is `develop`
- [x] I have run `make fmt` to format my code
- [x] I have run `make test` to run local tests
- [x] I have added tests that prove my fix is effective or that my feature works

### CODE REVIEW

**Comment by lizining1231** [comment]
CI 的-race job 报的是具有偶发性的全局 common/config.Environment.dynamicConfiguration 字段的数据竞争，或与本改动无关。以下是初步排查与确定：

### 报错日志
[ci-race.txt](https://github.com/user-attachments/files/31082289/ci-race.txt)
已尝试本地难以稳定用-race复现（`-race` 单测 ×单轮/多轮、全包 `-race`、`GOMAXPROCS=2` 组合压测均通过），竞态窗口或与时序强相关，尝试构造A/B测试成功复现。

### 问题
CI `-race` job 报 `registry/directory` 包内**全局 `common/config.Environment.dynamicConfiguration` 字段**的数据竞争：

- **写**：测试 goroutine（goroutine 40）`TestNewRegistryDirectoryConsumerListenerUsesResolvedApplicationName` 调用 `SetDynamicConfiguration`（directory_test.go:195）
- **读**：**跨测试遗留**的订阅 goroutine（goroutine 36，ID < 40）在 `TestTypedNilCachedInvokerDoesNotPanic` 遗留的事件处理中经 `tag.PriorityRouter.Notify` 调用 `GetDynamicConfiguration`（router.go:108）

### 原因

1. **无锁全局单例字段**：environment.go:41 的 `dynamicConfiguration` 是普通字段，Set/Get 无同步
2. **订阅 goroutine 跨测试存活**：mock_registry.go:99 起的 goroutine 阻塞在 `listener.Next()`（无缓冲 channel）上永不退出；`Destroy` 仅置标志、不关闭 channel
3. **测试无清理**：directory_test.go:573 的 `normalRegistryDir` 无 `t.Cleanup`

### 时序证明

`09.761Z` 第 3 次 Subscribe 遗留 goroutine 处理 TEST2 事件：
`L276 update begin → L304 selector add → L305 tag 日志`（仅 `GetDynamicConfiguration` 返回 nil 才打印）为**同一 goroutine 的同步执行链**，即读方，其与写方测试的写交错触发 DATA RACE

**Comment by lizining1231** [comment]
已提issue #3664 ，并同步更新race issues清单 #3614

**Review by Alanxtl — APPROVED** [review]

**Review by AlexStocks — APPROVED** [review]

**Comment by AlexStocks** [comment]
the ci failed.

File: protocol/triple/triple_protocol/client_stream.go [hunk 1]
```diff
@@ -72,7 +72,8 @@ func (c *ClientStreamForClient) Send(request any) error {
 }
 
 // CloseAndReceive closes the send side of the stream and waits for the
-// response.
+// response. If closing the send side or receiving the response fails, the
+// receive side is closed before the error is returned.
```
[the remark this answers is not in the document]
Comment (lizining1231) [reply protocol/triple/triple_protocol/client_stream.go]: Thanks. Updated the comment to say `CloseResponse` is called on a best-effort basis before the error is returned, matching the implementation that discards the error.

File: protocol/triple/triple_protocol/handler_stream_test.go [hunk 2]
```diff
@@ -0,0 +1,158 @@
...
+	"errors"
+	"io"
+	"net/http"
+	"testing"
+)
+
+import (
+	"dubbo.apache.org/dubbo-go/v3/protocol/triple/triple_protocol/internal/assert"
+	pingv1 "dubbo.apache.org/dubbo-go/v3/protocol/triple/triple_protocol/internal/gen/proto/connect/ping/v1"
+)
```
[the remark this answers is not in the document]
Comment (lizining1231) [reply protocol/triple/triple_protocol/handler_stream_test.go:30]: Thanks for flagging this
I checked and `gofmt` actually keeps separate import blocks unchanged (`gofmt -d` on this file produces no diff), and the repo consistently uses separate blocks to group stdlib vs internal imports (e.g. `protocol_grpc.go`, `handler_stream_compat.go`). So I left it as-is to match the existing convention.

File: protocol/triple/triple_protocol/client_stream_test.go [hunk 3]
```diff
@@ -107,3 +108,197 @@ type nopStreamingClientConn struct {
...
+	responseHdr  http.Header
+	responseTrlr http.Header
+
+	sendCalls      int
+	receiveCalls   int
+	closeReqCalls  int
+	closeRespCalls int
+}
+
+func (c *stubStreamingClientConn) Send(any) error {
```
[the remark this answers is not in the document]
Comment (lizining1231) [reply protocol/triple/triple_protocol/client_stream_test.go:135]: Thanks. Added explicit `Spec()` and `Peer()` zero-value implementations on `stubStreamingClientConn`, matching `stubHandlerConn` and avoiding the nil embedded-interface panic.

**Comment by lizining1231** [comment]
> the ci failed.

done.
